src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copiedStateObject = { ...state };
+  const newState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+
+        Object.assign(copiedStateObject, action.extraData);
+        break;
+
+      case 'removeProperties':
```

Comment:
Consider will be better if you create a variable for all magic words and them everywhere, it's a good practice